app/lib/features/tasks/providers/tasklists_providers.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:acter/features/tasks/providers/notifiers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:riverpod/riverpod.dart';
+
+//Single Task List Item based on the task list id
+final taskListItemProvider =
+    AsyncNotifierProvider.family<TaskListItemNotifier, TaskList, String>(
+  () => TaskListItemNotifier(),
+);
+
+//List of all task list
+//If space Id given then it will return list of space task list
+final tasksListsProvider = AsyncNotifierProvider.family<AsyncTaskListsNotifier,
+    List<TaskList>, String?>(() => AsyncTaskListsNotifier());
```

Comment:
this way we only hold one in-memory list and only one signaling provider to the underlying SDK.

```suggestion
//List of all task list
//If space Id given then it will return list of space task list
final allTasksListsProvider = AsyncNotifierProvider<AsyncTaskListsNotifier,
    List<TaskList>>(() => AsyncAllTaskListsNotifier());
    
final taskListProvider = FutureProvider.family<List<TaskList>, String?>((ref, spaceId) async {
  final allTaskLists = await ref.watch(allTasksListsProvider.future);
  if (spaceId == null) {
    return allTaskLists;
  } else {
    return tasksList
      .where((t) => t.spaceIdStr() == spaceId!)
      .toList(); // this might throw internally
   }
});
```